core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -414,9 +425,54 @@ private[spark] class TaskSchedulerImpl(
             launchedAnyTask |= launchedTaskAtCurrentMaxLocality
           } while (launchedTaskAtCurrentMaxLocality)
         }
+
         if (!launchedAnyTask) {
-          taskSet.abortIfCompletelyBlacklisted(hostToExecutors)
-        }
+          taskSet.getCompletelyBlacklistedTaskIfAny(hostToExecutors) match {
+            case taskIndex: Some[Int] => // Returns the taskIndex which was unschedulable
+              if (conf.getBoolean("spark.dynamicAllocation.enabled", false)) {
+                // If the taskSet is unschedulable we kill the existing blacklisted executor/s and
+                // kick off an abortTimer which after waiting will abort the taskSet if we were
+                // unable to get new executors and couldn't schedule a task from the taskSet.
+                // Note: We keep a track of schedulability on a per taskSet basis rather than on a
+                // per task basis.
+                if (!unschedulableTaskSetToExpiryTime.contains(taskSet)) {
+                  hostToExecutors.valuesIterator.foreach(executors => executors.foreach({
+                    executor =>
+                      logDebug("Killing executor because of task unschedulability: " + executor)
+                      blacklistTrackerOpt.foreach(blt => blt.killBlacklistedExecutor(executor))
```

Comment:
Seriously? You killed all executors ? What if other taskSets' tasks are running on them ?

BTW, if you want to refresh executors, you have to enable `spark.blacklist.killBlacklistedExecutors`  also.

Reply:
- To refresh executors, you need to enable `spark.blacklist.killBlacklistedExecutors`.
- I was thinking about it, killing all the executors is a little too harsh. Killing only a single executor  would help mitigate this, although this would also lead to failing the running tasks on that executor.